test/e2e/deployment.go

```diff
@@ -433,6 +446,8 @@ func testRolloverDeployment(f *framework.Framework) {
 
 	// Wait for replica set to become ready before adopting it.
 	framework.Logf("Waiting for pods owned by replica set %q to become ready", rsName)
+	By("Wait for pods become ready")
+	Expect(framework.WaitForPodsReady(c, ns, podName, 10)).NotTo(HaveOccurred())
```

Comment:
Add this is to make sure the pods are ready in case of the WaitForReadyReplicaSet doesn't work, if the pods are ready and the WaitForReadyReplicaSet failed, we can know that WaitForReadyReplicaSet is fragile.